bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java

```diff
@@ -426,38 +467,160 @@ private void runPanelDiscovery() {
         }
     }
 
+    /**
+     * This is based on the touch event detection described in https://forum.nanoleaf.me/docs/openapi#_842h3097vbgq
+     */
+    private static boolean touchJobRunning = false;
+    private void runTouchDetection() {
+        if (touchJobRunning) {
+            logger.debug("touch job already running. quitting.");
+            return;
+        }
+        try {
+            touchJobRunning = true;
+            URI eventUri = OpenAPIUtils.getUri(getControllerConfig(),API_EVENTS,"id=4");
+            logger.debug("touch job registered on: {}",eventUri.toString());

```

Comment:
You don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.